### ISSUE TITLE
Handle <slot> elements

### ISSUE THREAD

**Issue by edoardocavazza** [chialab/dna#31]
DNA components can only render their own templates, but sometimes some contents need to be inject for their parent. At the moment, this is possible passing a function property to the child element:

**Host template**
```js
const inject = <h1>Hello</h1>

<child-element title={inject}></child-element>
``` 

**Child template**
```js
<div>{this.title}</div>
```

The problem with this approach is that the developer could easily lose the hierarchy of the template tree.

A better implementation could replicate the slot element specification of ShadowDOM v1. The components template will look like:

**Host template**
```js
<child-element><h1>Hello</h1></child-element>
```
**Child template**
```js
<div><slot></slot></div>
```

**Comment by edoardocavazza** [comment]
More informations here: https://developers.google.com/web/fundamentals/web-components/shadowdom#composition_slot

**Comment by edoardocavazza** [comment]
Released in v3.0.0-alpha.1
